lib/internal/Magento/Framework/App/Response/Http.php

```diff
@@ -131,7 +133,7 @@ public function sendVary()
      */
     public function setPublicHeaders($ttl)
     {
-        if ($ttl < 0 || !preg_match('/^[0-9]+$/', $ttl)) {
+        if ($ttl < 0 || !preg_match('/^[0-9]+$/', (string) $ttl)) {
```

Comment:
@mrtuvn after discussing this change internally, decided that it makes sense to get rid of this `preg_match`. This way, we won't need the `string` casting.
```suggestion
        if ($ttl < 0 || !is_numeric($ttl)) {
```

Reply:
Thanks you just corrected again